### DISCUSSION TITLE
Solweig landcover scheme questions

### DISCUSSION THREAD

**Discussion by paolodep36** [UMEP-dev/UMEP#632] [Q&A]
Hi guys, thank you for the support you always show. I am trying to study well how the solweig model works and I have some questions for you. I noticed that the albedo and emissivity values of a scene are fixed for the reference landcover. I had a chance to open the landcoverclasses2016.txt file and noticed values of albedo, emissivity. Tstart and Tmax. I have doubts about how these values can work for every scene in the world. I have rasters of albedo, landcover and landsurface temperatures available for every city of he world.  Can I somehow use them in the solweig model or update the landcoverclasses file according to a hypothetical rule?
Can you suggest a guide or something to study to know more about what i am creating with the Solweig model? what happens if the landcover scheme is not used?

**Comment by biglimp** [comment]
Thanks for your nice comments. The values in landcoverclasses2016.txt can indeed be changed according to the specific environment you want to investigate. The values are derived form various empirical data that we have gathered over the years. IT is also possible to add more ground cover classes by just add a new line in the file. That is probably what I recommend you to do if you want to make changes. Just a heads-up, Do not use values 3 and 4 as that are occupied be different kinds of trees.

If you dont use and land cover scheme, the default values from Lindberg et al. 20087 is used for all ground pixels.

On this note, we are currently working on an update of the model for wall surface temperatures where we use a semi-infinite material approach and acctualy calulate surface temperatures based on radiation fluxes in and out for each wall voxel. This will also improve nighttime values for walls. Work in progress by @nilswallenberg. We are also thinking on adding something new for ground surface temperature. If you want to join that tast or have any good ideas, You are very welcome to join. /Fredrik

**Reply by paolodep36** [reply]
@biglimp Thank you for your answer. Can you provide me a link of the doi?. Anyways. what i was trying to say is that i have raster of albedo, land surface temperature and potentially raster of emissivity(from Aster). Is there a way to use these rasters to compute the solweig model? From what i understand now the model use fixed values for the whole scene, but in the reality albedo and land surface temperature values can vary dinamically for every pixel.  I can obviously adapt the landcoverclasses file doing zonal statistics for each landcover but how i derive Ts_deg Tstart TmaxLST ? How the values now reported in the landcover scheme works for the winter season and the summer season if they are fixed? This could help me to understand how the Solweig model works! Thanks in advance

**Reply by biglimp** [reply]
Yes, theoretically, it would be possible to implement separate grid layers for different parameters such as albedo but that would require some re-coding of the model. The links to SOLWEIG-related papers are found here: https://umep-docs.readthedocs.io/en/latest/Introduction.html#evaluation-and-application-studies

**Reply by paolodep36** [reply]
Hi @biglimp it's been a while from my last question. But i would like to have a code or a methodology more clear to reconstruct  the landcover_classes file embedded in solweig to create my own classes. i have access to surface temperature , albedo, emissivity and lancover rasters and i want to access to air temperature from era5(even if the resolution is not the best). how can i derive Ts_deg Tstart TmaxLST ? This will help me model each parameters of the classes i have in my landcover. I really need it. Thanks in advance as always.

**Reply by biglimp** [reply]
Ts_deg and Tstart is derived according to the methodology presented in Lindberg et al. 2008 IJBM and Lindberg et al. 2016 IJBM. TmaxLST is decided on common knowledge where materials with higher thermal capacity should have a later time e.g.3 pm that e.g. grass (2 pm).

**Comment by Moyekoli24** [comment]
Dear @biglimp,
I'm delighted to hear that a new model for calculating Ts is under construction. 

I'd like to follow up on this. If I have surface temperature measurements for different coatings for a period of one month, is it possible to reconstruct a landcover by doing a regression (Ts-Ta) with the time when the sun elevation is at its maximum (Nmax) ? I think that's what's explained in the article, but it's not clear about the period and the time step. Should I do a regression with each Nmax per day or is it the monthly Nmax?

**Reply by biglimp** [reply]
Yes, you can. Take out all of your clear days and make a regression line. Maybe your time is a bit too short but you can try.
